src/infra/azure/functions/put-signature-request-status.ts

```diff
@@ -0,0 +1,89 @@
+import { AzureFunction } from "@azure/functions";
+
+import { pipe, flow } from "fp-ts/lib/function";
+import { sequenceS } from "fp-ts/lib/Apply";
+import * as RE from "fp-ts/lib/ReaderEither";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as E from "fp-ts/lib/Either";
+
+import { createHandler } from "@pagopa/handler-kit";
+import {
+  success,
+  HttpRequest,
+  path,
+  error,
+  body,
+} from "@pagopa/handler-kit/lib/http";
+import * as azure from "@pagopa/handler-kit/lib/azure";
+
+import { validate } from "@pagopa/handler-kit/lib/validation";
+import { requireSubscriptionId } from "../../http";
+
+import { Subscription } from "../../../signature-request/subscription";
+
+import {
+  SignatureRequest,
+  SignatureRequestId,
+  SignatureRequestStatus,
+} from "../../../signature-request/signature-request";
+import {
+  getSignatureRequest,
+  upsertSignatureRequest,
+} from "../cosmos/signature-request";
+import { SignatureRequestDetailView } from "../../api-models/SignatureRequestDetailView";
+import { updateStatusRequestSignature } from "../../../app/use-cases/request-signature";
+import { PutSignatureStatusBody } from "../../api-models/PutSignatureStatusBody";
+
+const updateStatus = updateStatusRequestSignature(
+  upsertSignatureRequest,
+  getSignatureRequest
+);
+
+export const requireSignatureRequestId: (
+  req: HttpRequest
+) => E.Either<Error, SignatureRequest["id"]> = flow(
+  path("signatureRequestId"),
+  E.fromOption(() => new Error("Missing signatureRequestId in path")),
+  E.chainW(validate(SignatureRequestId, "Invalid signatureRequestId in path"))
```

Comment:
I'm not sure we should use the interal model type instead of the API (generated) one. See the following comment.